web/pgadmin/misc/properties/CollectionNodeProperties.jsx

```diff
@@ -133,7 +133,9 @@ export function CollectionNodeView({
         _type: row.original._type,
       }));
     } else {
-      selRows = selRowModels.map((row) => row.original.oid);
+      let nodeObj = pgBrowser.Nodes[itemNodeData?._type.replace('coll-', '')];
+      let schema = nodeObj.getSchema?.call(nodeObj, treeNodeInfo, itemNodeData);
```

Comment:
`schema` is already been set at one more place. Use the same schema obj. Do not recreate the logic.

Reply:
The only place schema is being used is inside the useEffect where table columns are being created, so I reused that logic. I'll check again if we have reference to schema at any other place which can be reused.

Reply:
The useEffect schema can be re-used. Create a reference to it using `const schemaRef = useRef()` and then `schemaRef.current = <code>`. Then you can use `schemaRef` directly.

Reply:
Please review now.